portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -32,41 +39,58 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-  private List<Comment> comments = new ArrayList<>();
-
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("application/json");
-    String json = new Gson().toJson(comments);
-    response.getWriter().println(json);
+    // Create Query for Datastore
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    List<Comment> comments = new ArrayList<>();
+    for (Entity entity : results.asIterable()) {
+      String username = (String) entity.getProperty("username");
+      String commentBody = (String) entity.getProperty("commentBody"); 
+      String timestamp = (String) entity.getProperty("timestamp");
+
+      Comment comment = new Comment(username, commentBody, timestamp);
+      comments.add(comment);
+    }
+
+    Gson gson = new Gson();
+
+    response.setContentType("application/json;");
+    response.getWriter().println(gson.toJson(comments));
   }
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     // Get the input from the form
-    String username = getParameter(request, "user-name", "").trim();
-    String commentText = getParameter(request, "text-input", "").trim();  
+    String username = getParameter(request, "user-name").trim();
```

Comment:
Nit: it might be better to filter empty username and / or commentBody.

Reply:
I actually put "required" in my `username` and `commentBody` input textarea. So the method `doPost` will not even be executed as if the fields are empty. But yeah I forgot to consider the case when the the `username` and/or `commentBody` are whitespaces! Will add this case in.

Reply:
You are right!
I misread the code. 😂